models/events.go

```diff
@@ -39,6 +44,15 @@ type CadenceEvents struct {
 
 // NewCadenceEvents decodes the events into evm types.
 func NewCadenceEvents(events flow.BlockEvents) (*CadenceEvents, error) {
```

Comment:
Why Event Streaming API returns a different order? 

Can we mention what was the default order? 

And have we thought about changing the order in the EventStreaming API?

Reply:
That's a good question. So the default order seems to be ascending, for the most part. I am not sure if the Event Streaming API does offer such a guarantee, on the received order of the events.

However, regardless of whatever fixes we chose to do on the Event Streaming API, there are blocks that have EVM-related events out-of-order, see: https://github.com/onflow/flow-evm-gateway/pull/622#issuecomment-2429274093.

And some other cases:
```bash
FlowTxIndex: 0, FlowEventIndex: 15
[EVM]:  A.e467b9dd11fa00df.EVM.TransactionExecuted(hash: [...], index: 0, ...)
FlowTxIndex: 0, FlowEventIndex: 16
[EVM]:  A.e467b9dd11fa00df.EVM.TransactionExecuted(hash: [...], index: 1, ...)
FlowTxIndex: 1, FlowEventIndex: 15
[EVM]:  A.e467b9dd11fa00df.EVM.TransactionExecuted(hash: [...], index: 0, ...)
```
This case though seems to be a race, as the `index` field of `EVM.TransactionExecuted` is not strictly increasing in the scope of an EVM block.

In order to incorporate the `offchain` package from `flow-go`, and replay EVM blocks & transactions to build a local state index, it is a prerequisite that we supply the blocks & transactions in the correct order, otherwise there will be state mismatches.

So the change in this PR, is more of a safety check, to ensure that we can mitigate such cases, no matter how the 3rd party tools, such as Event Streaming API, behave.